LeaseToOwn.lua

```diff
@@ -9,118 +9,95 @@
 
-    if itemsList ~= nil and itemsList.totalItemCount == 0 then
-        logger.info("LeaseToOwn: page itemList (is nil = "..tostring(itemsList ~= nil)..") is empty. Abort!")
-        return
+    local menuButton = g_inGameMenu.leasePurchase_Button
+
+    if menuButton ~= nil then
+        menuButton:unlinkElement()
+        menuButton:delete()
+        g_inGameMenu.leasePurchase_Button = nil
     end
+end
 
-    local selectedItemIdx = itemsList.selectedIndex
-    if selectedItemIdx < 1 then
-        logger:info("LeaseToOwn: selected item idx "..selectedItemIdx.." is invalid. Abort!")
-        return
+function LeaseToOwn:onIndexChanged(index, count)
+    if g_inGameMenu.leasePurchase_Button ~= nil then
+        local vehicle = g_inGameMenu.vehiclesList.dataSource.vehicles[index].vehicle
+        if vehicle:getPropertyState() == VehiclePropertyState.LEASED then
+            g_inGameMenu.leasePurchase_Button:setDisabled(false)
+        else
+            g_inGameMenu.leasePurchase_Button:setDisabled(true)
+        end
     end
-	
-	local vehicle = itemsList.dataSource.displayItems[selectedItemIdx].concreteItem
-	if vehicle ~= nil then
-		local isPageListOfLeasedVehicles = (vehicle.propertyState == Vehicle.PROPERTY_STATE_LEASED)
-		if isPageListOfLeasedVehicles and shopMenu.leasePurchase_Button == nil then
-			local leasePurchaseElement = shopMenu.menuButton[1]:clone(self)
-			leasePurchaseElement:setText(g_i18n:getText("LeaseToOwn_PURCHASE"))
-			leasePurchaseElement:setInputAction("MENU_EXTRA_1")
-			leasePurchaseElement.onClickCallback = function ()
-				LeaseToOwn.purchase()
-			end
-
-			shopMenu.menuButton[1].parent:addElement(leasePurchaseElement)
-			shopMenu.leasePurchase_Button = leasePurchaseElement
-		end
-	end
 end
 
 function LeaseToOwn:purchase()
     logger:debug("LeaseToOwn:purchase leasePurchaseElement callback begin")
 
-    local shopMenu = g_currentMission.shopMenu
-    local currentPage = shopMenu.pageShopItemDetails
-    local itemsList = currentPage.itemsList
-    local selectedItemIdx = itemsList.selectedIndex
-    local vehicle = itemsList.dataSource.displayItems[selectedItemIdx].concreteItem
-
+    local selectedVehicleIndex = g_inGameMenu.vehiclesList.dataSource.vehicles[g_inGameMenu.vehiclesList.selectedIndex]
+    local vehicle = selectedVehicleIndex.vehicle
     local price = calculateVehicleValue(vehicle)
+
     LeaseToOwn.price = price
     LeaseToOwn.selectedVehicle = vehicle
 
     local text = string.format(g_i18n:getText("LeaseToOwn_leasePurchaseQuestion"),
                                vehicle:getName(),
                                g_i18n:formatMoney(price))
 
-    g_gui:showYesNoDialog({text = text,
-                           title = g_i18n:getText("LeaseToOwn_leasePurchaseQuestionHeader"),
-                           callback = LeaseToOwn.onConfirm,
-                           target = LeaseToOwn})
+    YesNoDialog.show(LeaseToOwn.onConfirm, LeaseToOwn, text)
```

Comment:
`showYesNoDialog` is no longer part of the `g_gui` API. GIANTS have instead moved to a mixin approach where `show` is implemented on each type of dialog.